### DISCUSSION TITLE
Plotter default camera roll value dependence with the focal point

### DISCUSSION THREAD

**Discussion by ifermion** [pyvista/pyvista#3972] [Q&A]
Hi,

It seemed to me that the default `plotter.camera.roll` value is linked to the value of camera's focal point `plotter.camera.focal_point`  when I was trying to understand how the camera roll is configured in Pyvista. I expected it to be 0° for all the Plotter objects by default. Can someone please explain why I'm seeing a non-zero roll value by default and its relation with `plotter.camera.focal_point` and maybe some other parameters in the `Plotter.camera` ?

Thanks in advance !

**Comment by banesullivan** [comment]
I'm no computer graphics or computer vision expert so this is where I tend to get lost. I recommend posting this question on the VTK discourse for the best insight as this is coming directly from VTK: https://discourse.vtk.org/

For what I understand, the camera rolls about its direction of projection. This direction `plotter.camera.direction` changes as the camera position changes. As for how the roll gets adjusted as that changes... oof my head starts to hurt. You may want to look into some of VTK's docs on the trackball interaction style which is mostly controlling all of this (and our default in PyVista): https://vtk.org/doc/nightly/html/classvtkInteractorStyleTrackballCamera.html#details
